opba-protocols/hbci-protocol-tests/hbci-bdd-sandbox/src/test/java/de/adorsys/opba/protocol/hbci/tests/e2e/sandbox/HbciSandboxPaymentAndTransactionsAfterE2EHbciProtocolTest.java

```diff
@@ -129,22 +177,38 @@ void testPaymentWithStatusWithScaAndAfterThatPaymentAppearsInTransactions() {
                 .user_max_musterman_selected_sca_challenge_type_push_tan_to_embedded_authorization()
                 .and()
                 .user_max_musterman_provided_sca_challenge_result_to_embedded_authorization_and_sees_redirect_to_fintech_ok();
-
         accountInformationResult
                 .open_banking_has_consent_for_max_musterman_account_list()
                 .fintech_calls_consent_activation_for_current_authorization_id()
                 .open_banking_can_read_max_musterman_hbci_transaction_data_using_consent_bound_to_service_session_with_extra_transactions(
-                        MAX_MUSTERMAN_BANK_BLZ_30000003_ACCOUNT_ID, BANK_BLZ_30000003_ID, DATE_FROM, DATE_TO, BOTH_BOOKING, new BigDecimal[] {new BigDecimal("-1.03")}
+                        MAX_MUSTERMAN_BANK_BLZ_30000003_ACCOUNT_ID, BANK_BLZ_30000003_ID, DATE_FROM, DATE_TO, BOTH_BOOKING, extraTransactions
+                );
+    }
+
+    private void getTransactions() {
+        getTransactions(new BigDecimal[]{}, false);
+    }
+
+    private void getTransactionsAndCheckThatPaymentAppearsInTransactions() {
+        getTransactions(new BigDecimal[]{new BigDecimal("-1.03")}, false);
+    }
+
+    void getTransactionsWithCacheUpdate() {
+        BigDecimal[] extraTransactions = {new BigDecimal("-1.03"), new BigDecimal("-1.03")};
+        getTransactions(extraTransactions, true);
+    }
+
+    private void checkPaymentsAreNotInTransactions() {
+        accountInformationResult
+                .open_banking_can_read_max_musterman_hbci_transaction_data_using_consent_bound_to_service_session(
+                        MAX_MUSTERMAN_BANK_BLZ_30000003_ACCOUNT_ID, BANK_BLZ_30000003_ID, DATE_FROM, DATE_TO, BOTH_BOOKING
                 );
     }
 
     private void makeHbciAdapterToPointToHbciMockEndpoints() {
         adapterProperties.getAdorsysMockBanksBlz().stream()
                 .flatMap(it -> bankProfileJpaRepository.findByBankBankCode(String.valueOf(it)).stream())
-                .map(it -> {
-                    it.setUrl("http://localhost:" + port + "/hbci-mock/");
-                    return it;
-                })
+                .peek(it -> it.setUrl("http://localhost:" + port + "/hbci-mock/"))
```

Comment:
Avoid using `peek` to modify the content of the stream,
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#peek-java.util.function.Consumer-

>API Note:
    This method exists mainly to support debugging, where you want to see the elements as they flow past a certain point in a pipeline: